### ISSUE TITLE
False positive Naming/MemoizedInstanceVariableName

### ISSUE THREAD

**Issue by alexandrz** [rubocop-hq/rubocop#9205]
```ruby
class Test
  def self.inherited(klass)
    klass.define_method(:values) do
      @values ||= do_something
    end
  end

  inherited(self)
end
```

## Expected behavior

Memoized variable @values matches method name `values`.

## Actual behavior

Rebocop reports an offence:

Naming/MemoizedInstanceVariableName: Memoized variable @values does not match method name inherited. Use @inherited instead.

## RuboCop version

```
bundle exec rubocop -V
1.6.1 (using Parser 2.7.2.0, rubocop-ast 1.3.0, running on ruby 2.6.6 x86_64-linux-musl)
```
